flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/IntermediateResultPartitionID.java

```diff
@@ -27,29 +28,76 @@
 /**
  * Id identifying {@link IntermediateResultPartition}.
  */
-public class IntermediateResultPartitionID extends AbstractID implements ResultID {
+public class IntermediateResultPartitionID implements Comparable<IntermediateResultPartitionID>, ResultID, java.io.Serializable {
```

Comment:
Why is `Comparable` needed?

Reply:
First of all, `AbstractID` implements this inferface. I do not remember which class compare the `IntermediateResultPartitionID` but I do encounter issue when I did not implement this at the first time. It may take me some time to find it.